partrt/partrt

```diff
@@ -253,20 +253,41 @@ get_arg()
     printf "%s" "$2"
 }
 
+# Get monotonic time. Useful to implement timeouts, which do not break when
+# the wall clock is shifted.
+get_monotonic()
+{
+    cut -d. -f1 /proc/uptime
+}
 
 # Writes values to file and does proper error checking
+# If writing fails with EBUSY, then keep retrying for 5s
 # $1 File
 # $2 Value
 write_to_file () {
-    local file=$1
-    local val=$2
+    local -ri start=$(get_monotonic)
+    local -r file="$1"
+    local -r val="$2"
 
-    if [ -e $file ]; then
-        echo $val > $file || exit_msg "Failed to write $val into $file"
-        verbose_printf "echo $val > $file"
-    else
+    if ! [ -e "$file" ]; then
         verbose_printf "$file: Does not exist"
+        return
     fi
+
+    while true; do
+        local err=$(LC_ALL=C echo 2>&1 "$val" > "$file")
+        if [ $? -eq 0 ]; then
+            verbose_printf "echo $val > $file"
+            return
+        fi
+        if ! echo "$err" | grep -q 'Device or resource busy'; then
+            exit_msg "Failed to write $val into $file: $err"
+        fi
+        if [ "$(get_monotonic)" -gt "$((start+5))" ]; then
```

Comment:
I'd like to have the hardcoded value in a variable. In the future we might want to supply a parameter to specify whether we want this timeout or not and if so, how much.

Reply:
Updated